testlapack/dlahqr.go

```diff
@@ -312,11 +312,11 @@ func testDlahqr(t *testing.T, impl Dlahqrer, test dlahqrTest) {
 		}
 	}
 
-	// Check that wr and wi have not been modified outside [start:ihi+1].
```

Comment:
Lapack's documentation is not consistent in places, so some routines document wr and wi to be of length `n`, some only `ihi+1`. I noticed this as I propagated upwards in the dependency tree and things had to be made to work together. Dhseqr is the first routine that writes into the whole wr and wi, Dlahqr and DlaqrX operate only on `[ilo:ihi+1]`, so recently I changed Dlahqr (and maybe others) to require wr and wi have length exactly `ihi+1`. I changed the documentation and the routine but forgot to remove the checking of the (empty) slice after `ihi`. I included it here to avoid juggling too many branches or forgetting about it.